File: ios/Approach/Sources/ModelsLibrary/Gear.swift
import Foundation

public enum Gear {}

extension Gear {
	public typealias ID = UUID
}

extension Gear {
	public enum Kind: String, Codable, Sendable, Identifiable, CaseIterable {
		case shoes
		case bowlingBall
		case towel
		case other

		public var id: String { rawValue }
	}
}

extension Gear {
	public struct Summary: Sendable, Identifiable, Codable, Hashable {
		public let id: Gear.ID
		public let name: String
		public let kind: Kind
		public let ownerName: String?
		public let avatar: Avatar.Summary

		public init(id: Gear.ID, name: String, kind: Kind, ownerName: String?, avatar: Avatar.Summary) {
			self.id = id
			self.name = name
			self.kind = kind
			self.ownerName = ownerName
			self.avatar = avatar
		}
	}
}

File: android/app/src/main/java/ca/josephroque/bowlingcompanion/core/model/Gear.kt
package ca.josephroque.bowlingcompanion.core.model

import androidx.compose.runtime.Immutable
import ca.josephroque.bowlingcompanion.utils.SortableByUUID
import java.util.UUID

@Immutable
data class GearListItem(
	override val id: UUID,
	val name: String,
	val kind: GearKind,
	val ownerName: String?,
//	val avatar: Avatar,
): SortableByUUID

enum class GearKind {
	SHOES,
	BOWLING_BALL,
	TOWEL,
	OTHER,
}